packages/user/src/menu.ts

```diff
@@ -115,6 +124,8 @@ export class UserMenu extends Menu {
     this.title.label = this._user.isReady ? name : '';
     this.title.icon = caretDownIcon;
     this.title.iconClass = 'jp-UserMenu-caretDownIcon';
+    this.title.dataset = { empty: 'true' };
+    this.dataset.empty = 'true';
```

Comment:
It sounds to me that the source of truth is `this.title.dataset`. So could we remove modifying `this.dataset` in the `UserMenu`?
Changing the title dataset will trigger a change signal on the menu that will trigger an update on the menubar. So this should not be needed.

https://github.com/jupyterlab/lumino/blob/6da4c5899de06b9ed857b0c4355592e12f8c3476/packages/widgets/src/title.ts#L338-L344

```suggestion
```

Reply:
Hey @fcollonval, thanks for reviewing this PR!

We need to add the attribute to both datasets because the menu and the title are two different elements on the DOM.
The title is added to the menu bar while the menu is added to `body`, and both have a small border to delimiter the menu. The title shows the border and changes the background in the menu bar to differentiate it from other menus. In contrast, the menu adds a border to the menu element, which is positioned `absolute` or `relative` (I do not remember correctly) below the top bar in the same position as the title.

Reply:
I tested it locally and I go it now. Without it you get a pixel glitches from the empty menu:

![image](https://user-images.githubusercontent.com/8435071/147229240-e04f656f-13c8-46b2-a76e-e9d2067306dc.png)

So I would still not do things like this. I would set pointer events to `none` on the menu bar item if it is empty. This will prevent the menu appearance.

> I update the suggestion accordingly